File: app/src/main/java/com/eam/investire/model/Movimentacao.java
package com.eam.investire.model;

import com.eam.investire.config.ConfiguracaoFirebase;
import com.eam.investire.helper.Base64Custom;
import com.eam.investire.helper.DateCustom;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.database.DatabaseReference;

import java.util.Objects;

public class Movimentacao {
    private String categoria;
    private String descricao;
    private String tipo;
    private double valor;
    private String key;

    public Movimentacao() {
    }

    public void setData() {
    }

    public String getCategoria() {
        return categoria;
    }

    public void setCategoria(String categoria) {
        this.categoria = categoria;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }

    public String getTipo() {
        return tipo;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }

    public double getValor() {
        return valor;
    }

    public void setValor(double valor) {
        this.valor = valor;
    }

    public String getKey() {
        return key;
    }

    public void setKey(String key) {
        this.key = key;
    }

    public void salvar(String dataEscolhida) {
        FirebaseAuth autenticacao = ConfiguracaoFirebase.getFirebaseAutenticacao();
        String idUsuario = Base64Custom.codificarBase64(Objects.requireNonNull(Objects.requireNonNull(autenticacao.getCurrentUser()).getEmail()));
        String mesAno = DateCustom.mesAnoDataEscolhida(dataEscolhida);
        DatabaseReference firebase = ConfiguracaoFirebase.getFirebaseDatabase();
        firebase.child("movimentacao")
                .child(idUsuario)
                .child(mesAno)
                .push()
                .setValue(this);
    }
}

File: app/src/main/java/com/eam/investire/model/Movimentacao.kt
package com.eam.investire.model

import com.eam.investire.config.ConfiguracaoFirebase.firebaseAutenticacao
import com.eam.investire.config.ConfiguracaoFirebase.firebaseDatabase
import com.eam.investire.helper.Base64Custom.codificarBase64
import com.eam.investire.helper.DateCustom.mesAnoDataEscolhida
import java.util.*

class Movimentacao {
    var categoria: String? = null
    var descricao: String? = null
    var tipo: String? = null
    var valor = 0.0
    var key: String? = null
    fun setData() {}
    fun salvar(dataEscolhida: String?) {
        val autenticacao = firebaseAutenticacao
        val idUsuario = Objects.requireNonNull(Objects.requireNonNull(autenticacao!!.currentUser)?.email)?.let { codificarBase64(it) }
        val mesAno = mesAnoDataEscolhida(dataEscolhida!!)
        val firebase = firebaseDatabase
        if (idUsuario != null) {
            firebase!!.child("movimentacao")
                    .child(idUsuario)
                    .child(mesAno)
                    .push()
                    .setValue(this)
        }
    }
}